plugins/inputs/slurm/slurm_test.go

```diff
@@ -0,0 +1,23 @@
+package slurm
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestURLs(t *testing.T) {
+	for _, url := range []string{"http://example.com:6820", "https://example.com:6820"} {
+		plugin := Slurm{
+			URL: url,
+		}
+		require.NoError(t, plugin.Init())
+	}
+
+	for _, url := range []string{"httpp://example.com:6820", "httpss://example.com:6820"} {
+		plugin := Slurm{
+			URL: url,
+		}
+		require.Error(t, plugin.Init())
+	}
+}
```

Comment:
I'd like to see some actual tests of the Gather functions. Something that takes in a struct and ensures the metric is created.

Reply:
Hi! These should have been added on fa4f3f24aade64c46cbb89122bdd38bdc73659b1.

Also, whilst writing the tests I ran into several `panic()`s because of typos and the pointer dereference taking place inside the `Gather*` functions. These dereferences had to be added because otherwise the assertions were comparing addresses of variables instead of variables themselves. Anyway, I decided to `recover()` from `nil` pointer dereference `panic()`s on `Gather()`, but another approach would be to explicitly check for `nil` before referencing a pointer.

If only concerned with the tests passing, one can also check whether the pointers (i.e. addresses) have been added as records to the accumulator instead of the values themselves. This 'works' because at some point in the marshalling of all the data `telegraf` dereferences the pointers itself so that it's data that makes it out instead of these addresses. I initially leveraged this implicitly marshalling to avoid having to either check for `nil` or `recover()` myself, but maybe this approach is better.

Anyway, I'll do whatever you feel is best.

Thanks a ton for your input!

Reply:
Thanks for the updates!

> another approach would be to explicitly check for nil before referencing a pointer.

This is the approach I would go with. We should never panic in Telegraf and recovery is a heavy cost thing to do.

Reply:
That should've been added on cfd6ce07cf4898bb5ff6613b813a3ce2be283a30.

Thanks a ton for your input!